fix: design improvements for log in screen for cal video

bandhan-majumder opened the pull request:
## What does this PR do?

- Fixes #21754
- Fixes CAL-5910

## Visual Demo (For contributors especially)

<img width="663" height="565" alt="image" src="https://github.com/user-attachments/assets/b9d8892b-6bfa-47e5-b7e4-ff6841f9d975" />
<img width="365" height="507" alt="image" src="https://github.com/user-attachments/assets/73b7f7fd-02a0-42b9-bbb2-bca14c9a3df9" />
<img width="625" height="484" alt="image" src="https://github.com/user-attachments/assets/b1073a65-4210-441d-96a9-1dc3ee01f065" />



### Light mode
<img width="598" height="478" alt="image" src="https://github.com/user-attachments/assets/7c0109fd-f057-4367-9049-3acdfdd65575" />
<img width="332" height="466" alt="image" src="https://github.com/user-attachments/assets/732dbf97-a374-47d7-8a99-fc5ccfe2408a" />
<img width="627" height="448" alt="image" src="https://github.com/user-attachments/assets/2a5fe253-088d-4fac-9344-8d7dee9bee03" />

A visual demonstration is strongly recommended, for both the original and new change **(video / image - any one)**.

#### Video Demo (if applicable):

https://www.loom.com/share/b5a321de1a624bfd8a6a1c5db011bdb7?sid=0614c919-22a4-40ae-8eef-e6e7ad933aba

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

By confirming a booking and going to the cal video link with having `displayLogInOverlay` as true

- Are there environment variables that should be set?
yes (DAILY_API_KEY)
- What are the minimal test data to have?
dailyco api key and two users
- Any other important info that could help to test that PR
hardcode `displayLogInOverlay` to true 

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Improved the log in overlay for Cal Video by updating the design, adding clearer guest and sign-in options, and refining text for better user experience.

- **UI Improvements**
  - Redesigned the log in overlay with separate sections for joining as a guest or signing in.
  - Added new text strings and input handling for guest names.
  - Improved accessibility and button states.

<!-- End of auto-generated description by cubic. -->

bandhan-majumder committed:
fix: design improvements for log in screen for cal video

Signed-off-by: bandhan-majumder <bandhanmajumder16@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=22359) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=22359) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=22359) it.</sub>

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
fix: type

Signed-off-by: bandhan-majumder <bandhanmajumder16@gmail.com>

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'fix-21754-design' of https://github.com/bandhan-majumder/cal.com into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

Devanshusharma2005 reviewed (dismissed):
Hey! It looks like the typecheck is failing could you please take a look and fix it?
Also, it would be great if you could attach screenshots of the changes in both dark and light mode, as well as across different screen sizes for better visual coverage. Thanks!

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
add: light mode

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

Devanshusharma2005 reviewed (commented):
@bandhan-majumder  
Thank you for the PR it looks good to me. Let's wait for @CarinaWolli review as well before moving forward.

CarinaWolli commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -227,49 +298,135 @@ function ProgressBar(props: ProgressBarProps) {
+
+          <div className="mt-5 space-y-4">
+            <div>
+              <h4 className="text-lg font-semibold text-black dark:text-white">{t("sign_in_to_cal_com")}</h4>
```
should be same font as "Join as Guest" `text-base`

CarinaWolli commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -227,49 +298,135 @@ function ProgressBar(props: ProgressBarProps) {
+            </div>
+
+            <Button
+              color="primary"
```
```suggestion
              color="secondary"
```

CarinaWolli commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -147,7 +209,16 @@ export default function JoinCall(props: PageProps) {
         )}
       </div>
-      {displayLogInOverlay && <LogInOverlay isLoggedIn={!!loggedInUserName} bookingUid={booking.uid} />}
+      {true && !isUserNameConfirmed && (
```
```suggestion
      {displayLogInOverlay && !isUserNameConfirmed && (
```

Udit-takkar reviewed (commented):
![Image](https://github.com/user-attachments/assets/8836ea3b-adee-4b5f-afd4-73f35217b697)

Padding is a bit off here. It's 24px in figma

Udit-takkar commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -42,88 +43,149 @@ export default function JoinCall(props: PageProps) {
+          callFrame.setUserName(userName);
+        }
 
-      if (overrideName) {
-        callFrame.setUserName(overrideName);
```
Another thing is if overrideName name is passed we don't want to display the login modal and ask for name anywhere

bandhan-majumder committed:
add suggestions 

added padding change and removed hard coded values

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

CarinaWolli commented:
@Udit-takkar can you do a final review please? 🙏

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
Hi @kart1ka can you please review this PR?

kart1ka commented:
> Hi @kart1ka can you please review this PR?

Yes, we will review it.

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

kart1ka commented:
Hi @bandhan-majumder, I still see the old login screen (see the attached video). Am I doing something wrong here?

https://cap.link/w1rasp90g6fve5y

bandhan-majumder commented:
@kart1ka the `displayLogInOverlay` have to be true. I had hard coded that to true while developing. This is what @Udit-takkar also suggested in the issues section. Then the new modal will be visible 😁

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
hi @kart1ka , is the modal appearing now ?!

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
hi @hariombalhara , @emrysal   . Can you please review this pr ?!

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
@CarinaWolli the new modal will only occur when the property `displayLogInOverlay` is true. By default it's false and that is why you are seeing the old modal and not the new one. Would you like me to provide the hardcode true code for testing purpose?

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

CarinaWolli commented:
Sorry, I thought I did. It's working now 🙏

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Update apps/web/modules/videos/views/videos-single-view.tsx

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>

bandhan-majumder committed:
Update common.json

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
remove isLoggedin from LogInOverlayProps

bandhan-majumder commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -147,7 +209,16 @@ export default function JoinCall(props: PageProps) {
-      {displayLogInOverlay && <LogInOverlay isLoggedIn={!!loggedInUserName} bookingUid={booking.uid} />}
+      {displayLogInOverlay && !isUserNameConfirmed && (
+        <LogInOverlay
+          isLoggedIn={!!loggedInUserName}
```
Sure. Removed it from props

CarinaWolli replied:
But when I tested it last time we showed the login dialog even when the user was already logged in. Is this fixed as well?

bandhan-majumder commented:
@CarinaWolli 
implemented the suggestions. Now it does not have isloggedin prop and explicit texts apart from translations

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
The changes are made. Please review it once more.

anikdhabal commented:
@bandhan-majumder does the description has the updated image?

bandhan-majumder commented:
As per the @CarinaWolli 's suggestions, I have;
updated the images, removed the explicit texts, added required translations and removed the isLoggedIn prop

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented on apps/web/modules/videos/views/videos-single-view.tsx:
```diff
@@ -225,51 +295,134 @@ function ProgressBar(props: ProgressBarProps) {
 export function LogInOverlay(props: LogInOverlayProps) {
   const { t } = useLocale();
-  const { isLoggedIn, bookingUid } = props;
-  const [open, setOpen] = useState(!isLoggedIn);
```
Yeah it was. As you asked, i removed the isloggedin prop. I have attached the screenshot. Can you once confirm if they are  the desired screens or not?? If not, should the secondone also appear if th second condition is true??
It's in the previous review of yours.. should I add th link

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
type fix

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

kart1ka reviewed (changes requested):
It works fine when no user session exists. But when a user is logged in an empty screen is displayed. Can you pls fix it?

https://cap.link/dt024xbkktfwchp

kart1ka commented:
> @kart1ka it's working in my case as I have also recorded that in my [loom](https://www.loom.com/share/79928077c348426397e397bacaa1466b) and also attached a screen shot below. I am unable to reproduce the white screen error. On that basis, I am making the pull as ready for review again. @CarinaWolli can you please check it once if it's the same for you or not. Would be very helpful
> 
> <img width="1920" height="1029" alt="image" src="https://github.com/user-attachments/assets/7065ccf9-baeb-4258-90b3-40149a086bc5" />

In my case the white screen appears when the new login overlay is toggled on. Your loom video does not show this scenario.

bandhan-majumder committed:
screen flash fixed

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
@kart1ka 

its fixed. New loom updated with all the scenarios;
https://www.loom.com/share/6ffa7e36c93a4f5aafbdc92d43e305c3

bandhan-majumder committed:
rabbit suggestions

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
rabbit suggestions

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder commented:
@kart1ka all the changes are made

kart1ka committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
flakey test

bandhan-majumder commented:
@kart1ka @anikdhabal @Udit-takkar 
can this pr get approved

kart1ka committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

kart1ka reviewed (changes requested):
the login flow is broken when I try to login from the meeting link. See the video attached.

https://cap.link/2jweebxx0f72sa4

Can you please check and fix it?

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
There is one more issue I am facing while leaving the call, the default redirect page is not show when I sign in through modal and gets redirected. There is something wrong with the state. There are also too many commits already here in this PR. Should I close this and open a new one or push the changes here directly?

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Update videos-single-view.tsx

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

kart1ka committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
@kart1ka 
It's ready for review already. If the above loom and the explanation looks good to u, please consider it reviewing.

bandhan-majumder commented:
@kart1ka WDYT

bandhan-majumder commented:
The local development is too slow. It's in the issue section now too [here](https://github.com/calcom/cal.com/issues/23104). That's taking time while signing up and upon double clicking, it's showing that.

bandhan-majumder committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder commented:
@kart1ka @Devanshusharma2005 @anikdhabal 
this PR fixes the issues. Can you please review it?

kart1ka committed:
Merge branch 'main' into fix-21754-design

kart1ka commented:
> @kart1ka @Devanshusharma2005 @anikdhabal this PR fixes the issues. Can you please review it?

I will review it today.

kart1ka committed:
Merge branch 'main' into fix-21754-design

bandhan-majumder committed:
add: deps

bandhan-majumder committed:
coderabbit

kart1ka reviewed (approved):
Thanks for your contribution and for your patience while this was pending. We really appreciate your efforts. Approving now!

bandhan-majumder commented:
@kart1ka Thanks for continuously giving feedback.

bandhan-majumder commented:
@CarinaWolli @anikdhabal 
can I get a review and approval from you too?
